src/components/DrugChartSlider/DrugChartSlider.jsx

```diff
@@ -0,0 +1,546 @@
+  ] = useState(false);
+  const [showStartTimePassedWarning, setShowStartTimePassedWarning] =
+    useState(false);
+  const [showSchedulePassedWarning, setShowSchedulePassedWarning] = useState(
+    []
+  );
+  const [showScheduleOrderWarning, setShowScheduleOrderWarning] =
+    useState(false);
+  const [showEmptyScheduleWarning, setShowEmptyScheduleWarning] =
+    useState(false);
+  const [showEmptyStartTimeWarning, setShowEmptyStartTimeWarning] =
+    useState(false);
+  const invalidTimeText24Hour = "Please enter in 24-hr format";
+  const invalidTimeText12Hour = "Please enter in 12-hr format";
+
+  const isInvalidTimeTextPresent = () => {
+    const screenContent = document.body.textContent;
+    const invalidTimeText = enable24HourTimers
+      ? invalidTimeText24Hour
+      : invalidTimeText12Hour;
+    return screenContent.includes(invalidTimeText);
+  };
+
+  const isTimePassed = (newTime) => {
+    const currentTime = moment();
+    const enteredTime = enable24HourTimers
+      ? moment(newTime, "HH:mm")
+      : moment(newTime, "hh:mm A");
+    return currentTime.isAfter(enteredTime);
+  };
+
+  const handleSchedule = (newSchedule, index) => {
+    const newScheduleArray = [...schedules];
+    newScheduleArray[index] = enable24HourTimers
+      ? newSchedule
+      : moment(newSchedule, "hh:mm A");
+    setSchedules(newScheduleArray);
+    if (!isInvalidTimeTextPresent()) {
+      setShowSchedulePassedWarning((prevScheduleWarnings) => {
+        const newSchedulePassedWarnings = [...prevScheduleWarnings];
+        newSchedulePassedWarnings[index] = isTimePassed(newSchedule);
+        return newSchedulePassedWarnings;
+      });
+    }
+  };
+
+  const areSchedulesInOrder = (allSchedule) => {
+    return allSchedule.every((schedule, index) => {
+      if (index === 0) return true;
+      const currentTime = moment(schedule, "HH:mm");
+      const prevTime = moment(allSchedule[index - 1], "HH:mm");
+      return currentTime.isAfter(prevTime);
+    });
+  };
+
+  const validateSchedules = async (schedules) => {
+    if (schedules.some((schedule) => schedule === "")) {
+      return { isValid: false, warningType: "empty" };
+    }
+
+    if (areSchedulesInOrder(schedules)) {
+      return { isValid: true, warningType: "" };
+    } else {
+      return { isValid: false, warningType: "passed" };
+    }
+  };
+
+  const handleScheduleWarnings = async () => {
+    const { isValid, warningType } = await validateSchedules(schedules);
+    setShowEmptyScheduleWarning(!isValid && warningType === "empty");
+    setShowScheduleOrderWarning(!isValid && warningType === "passed");
+    return { isValid, warningType };
+  };
+
+  const isValidSchedule = async () => {
+    const { isValid, warningType } = await handleScheduleWarnings();
+    if (!isValid && (warningType === "empty" || warningType === "passed"))
+      return false;
+    return true;
+  };
+
+  const updateStartTimeBasedOnFrequency = (frequency, time) => {
```

Comment:
move all these methods out of the component to the DrugChartSlider utility file, to make the component more maintainable and readable. Please check which all methods can be moved to utility file